contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -12,37 +12,63 @@ import { IVault } from "../interfaces/IVault.sol";
 import { IOracle } from "../interfaces/IOracle.sol";
 import { IStrategy } from "../interfaces/IStrategy.sol";
 import { IUniswapV2Router } from "../interfaces/uniswap/IUniswapV2Router02.sol";
+import { IUniswapV3Router } from "../interfaces/uniswap/IUniswapV3Router.sol";
+import { IBalancerVault } from "../interfaces/balancer/IBalancerVault.sol";
+import { ICurvePool } from "../strategies/ICurvePool.sol";
 import "../utils/Helpers.sol";
 
 abstract contract BaseHarvester is Governable {
     using SafeERC20 for IERC20;
     using SafeMath for uint256;
     using StableMath for uint256;
 
-    event UniswapUpdated(address _address);
-    event SupportedStrategyUpdate(address _address, bool _isSupported);
+    enum SwapPlatform {
+        UniswapV2Compatible,
+        UniswapV3,
+        Balancer,
+        Curve
+    }
+
+    event SupportedStrategyUpdate(address strategyAddress, bool isSupported);
     event RewardTokenConfigUpdated(
-        address _tokenAddress,
-        uint16 _allowedSlippageBps,
-        uint16 _harvestRewardBps,
-        address _uniswapV2CompatibleAddr,
-        uint256 _liquidationLimit,
-        bool _doSwapRewardToken
+        address tokenAddress,
+        uint16 allowedSlippageBps,
+        uint16 harvestRewardBps,
+        SwapPlatform platform,
+        address swapRouterAddr,
+        bytes swapData,
+        uint256 liquidationLimit,
+        bool doSwapRewardToken
+    );
+    event RewardTokenSwapped(
+        address indexed rewardToken,
+        address indexed swappedInto,
+        SwapPlatform platform,
+        uint256 amountIn,
+        uint256 amountOut
+    );
+    event RewardProceedsTransferred(
+        address indexed token,
+        address farmer,
+        uint256 protocolShare,
+        uint256 farmerShare
     );
+    event RewardProceedsAddressChanged(address newProceedsAddress);
 
     // Configuration properties for harvesting logic of reward tokens
     struct RewardTokenConfig {
         // Max allowed slippage when swapping reward token for a stablecoin denominated in basis points.
         uint16 allowedSlippageBps;
         // Reward when calling a harvest function denominated in basis points.
         uint16 harvestRewardBps;
-        /* Address of Uniswap V2 compatible exchange (Uniswap V2, SushiSwap).
-         */
-        address uniswapV2CompatibleAddr;
+        // Address of compatible exchange router (Uniswap V2, SushiSwap).
+        address swapRouterAddr;
```

Comment:
with Balancer this is a BalancerVault and with Curve this is a Curve pool. Would a `swapPlatform` or `swapProtocol` or `swapProtocolEntryPoint` be more accurate?

Reply:
Since we already use `platform` for the `SwapPlatform` enum, something like `swapPlatformAddr` would be consistent ig, will make this change. Also, changing the `platform` to `swapPlatform` to avoid any confusion

Reply:
`swapPlatformAddr` is great